shell/browser/api/electron_api_web_request.cc

```diff
@@ -732,6 +778,47 @@ void WebRequest::HandleSimpleEvent(SimpleEvent event,
   info.listener.Run(gin::ConvertToV8(isolate, details));
 }
 
+void WebRequest::OnAuthRequiredListenerResult(uint64_t id,
```

Comment:
How about to use return value of `listener` instead of a `callback's callback` to set the value? (Maybe need make the listener async)

Reply:
@reitowo i did this mostly for consistency with the rest of `webRequest`!